### PR TITLE
config: moving grpc subscriptions to extensions

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: n/a
Docs Changes: inline
Release Notes: n/a

### CODE REVIEW

**Comment by alyssawilk** [comment]
Finally ready for review :-)

**Review by RyanTheOptimist — DISMISSED** [review]
Wow, impressive!

**Comment by alyssawilk** [comment]
@lizan ping for API review?

**Review by adisuissa — COMMENTED** [review]
Thanks! Left an API comment.

File: api/envoy/config/core/v3/config_source.proto [hunk 1]
```diff
@@ -281,15 +281,3 @@ message ExtensionConfigSource {
   // if they do not match any type URL in the set.
   repeated string type_urls = 4 [(validate.rules).repeated = {min_items: 1}];
 }
-
-// A placeholder proto for the registration for the REST subscription implementation.
-message RestSubscription {
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply api/envoy/config/core/v3/config_source.proto:286]: I can't imagine anyone using them - they were added as placeholders, there was literally no place in the config to use them.

**Review by adisuissa — COMMENTED** [review]
Thanks,
/lgtm api

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**